test/checks/aria/has-widget-role.js

```diff
@@ -0,0 +1,452 @@
+describe('has-widget-role', function () {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var node;
+	var checkContext = axe.testUtils.MockCheckContext();
+
+
+	afterEach(function () {
+		node.innerHTML = '';
+		checkContext._data = null;
+	});
+
+	it('should return false for elements with no role', function() {
```

Comment:
For future reference, it's better if you use unit tests to check if the logic of your code. Most of these tests just check that the ARIA lookup table is complete. It's easier and better to do that in the implementation tests. You can leave it as is, either way it's tested, but as a rule of thumb it's a good one to keep in mind.

Reply:
i agree. these tests were useful while adding this feature, don't think they need to stay. leaving for now pending the discussion of https://github.com/dequelabs/axe-core/issues/672